### PR TITLE
Fix Ruby runtime path

### PR DESCRIPTION
**Pull request by uilianries**
- Add relative path support for runtime
- Use CentOS6 when building on Linux

related issue: https://github.com/bincrafters/community/issues/721

### CODE REVIEW

File: test_package/conanfile.py [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 import os
-from distutils.spawn import find_executable
 from conans import ConanFile
 
 
 class RubyInstallerTestConan(ConanFile):
+    settings = "os", "arch", "build_type", "arch"
+    generators = "txt"
+
+    def  build(self):
```
Comment (SSE4) [inline test_package/conanfile.py]: extra space

File: conanfile.py [hunk 2]
```diff
@@ -1,79 +1,91 @@
...
+            autotools.install()
+            tools.rmdir(os.path.join(self.package_folder, "share"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_id(self):
+        del self.info.settings.compiler
 
     def package_info(self):
-        self.env_info.path.append(os.path.join(self.package_folder, "bin"))
+        self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
```
Comment (SSE4) [inline conanfile.py]: add message here

File: conanfile.py [hunk 3]
```diff
@@ -1,79 +1,91 @@
...
 
     def build_requirements(self):
-        if tools.os_info.is_windows:
+        if self.settings.os_build == "Windows":
             self.build_requires("7z_installer/1.0@conan/stable")
 
     def source(self):
-        if not tools.os_info.is_windows:
-            tools.get("https://cache.ruby-lang.org/pub/ruby/{}/{}.tar.gz".format(
+        if self.settings.os_build != "Windows":
```
Comment (SSE4) [inline conanfile.py]: don't use conditions within source method

File: conanfile.py [hunk 4]
```diff
@@ -1,79 +1,91 @@
...
+# -*- coding: utf-8 -*-
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
 
 
 class RubyInstallerConan(ConanFile):
     name = "ruby_installer"
     version = "2.3.3"
-    api_version = "2.3.0"
     license = "MIT"
```
Comment (SSE4) [inline conanfile.py]: is it MIT or GPL? package_info copies GPL file

Comment (uilianries) [reply conanfile.py]: oh, that's a good question, I need to check. good catch

Comment (uilianries) [reply conanfile.py]: Ruby license: https://spdx.org/licenses/Ruby.html

File: test_package/conanfile.py (hunk 1 above)
Comment (uilianries) [reply test_package/conanfile.py]: done

File: conanfile.py (hunk 3 above)
Comment (uilianries) [reply conanfile.py]: done

File: conanfile.py (hunk 4 above)
Comment (uilianries) [reply conanfile.py]: done

File: conanfile.py (hunk 2 above)
Comment (uilianries) [reply conanfile.py]: done

**Pull request merged**